### PR TITLE
Watch mode (--interval)

### PR DESCRIPTION
**Pull request by wookayin**
Adds `-i` / `--interval` to support built-in watch feature.

Based on @drons 's PR #30.

### CODE REVIEW

File: gpustat/core.py [hunk 1]
```diff
@@ -379,6 +379,8 @@ def print_formatted(self, fp=sys.stdout, force_color=False, no_color=False,
                         show_cmd=False, show_user=False, show_pid=False,
                         show_power=None, gpuname_width=16,
                         show_header=True,
+                        eol_char='\n',
```
Comment (Stonesjtu) [inline gpustat/core.py]: Is this eol character portable for OS other than POSIX? I think windows use '\n\r' or '\r\n'. However, it doesn't matter if we target on only POSIX systems.

File: gpustat/__main__.py [hunk 2]
```diff
@@ -66,7 +71,27 @@ def main(*argv):
                         version=('gpustat %s' % __version__))
     args = parser.parse_args(argv[1:])
 
-    print_gpustat(**vars(args))
+    if args.interval > 0:
```
Comment (Stonesjtu) [inline gpustat/__main__.py]: It should be better to have a minimal interval of 0.1 sec like `watch`.

File: gpustat/core.py (hunk 1 above)
Comment (Stonesjtu) [reply gpustat/core.py]: `os.linesep` can be a better choice of default parameter.

Comment (wookayin) [reply gpustat/core.py]: Yeah, it was previously just '\n'. I would say POSIX systems are just targetted, but totally fine with using `os.linesep`.

File: gpustat/__main__.py (hunk 2 above)
Comment (wookayin) [reply gpustat/__main__.py]: Good idea.

**Comment by Stonesjtu** [comment]
The doc needs an update too.

**Comment by Stonesjtu** [comment]
LGTM, but the README needs an update to show-off new option.

**Comment by wookayin** [comment]
Merged into master.

**Pull request merged**